canonical-web-and-design/maas-ui
# feat(ui): build machine configuration form

## Done

- Added components for machine configuration form
- Moved `TagLinks` component higher up the tree so it could be used in any machine page (incl here)
- Added `editable` prop to `FormikForm` which hides the buttons when false

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the react machine config form of a machine you can edit (i.e. is not locked and you are an admin) by navigating to `/MAAS/r/machine/<system_id>/configuration`
- Check that the form is disabled by default and the initial values are correct
- Click "Edit", change some of the values then submit
- Check that the form returns to disabled state and the values have changed
- Refresh the page to check that the changes are permanent
- Note that the `TagSelector` makes the page look broken when expanding below the viewport. I'll address that in a styling improvement follow-up.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2222
